lua/treesitter-context.lua

```diff
@@ -132,7 +143,39 @@ function M.enable()
 
   attached[cbuf] = true
 
-  autocmd({ 'WinScrolled', 'BufEnter', 'WinEnter', 'VimResized' }, update)
+  autocmd({ 'BufEnter', 'WinScrolled', 'VimResized' }, update)
+  autocmd({ 'WinEnter' }, function()
+    vim.schedule(update)
+  end)
+
+  autocmd({ 'WinResized' }, function()
+    local event = vim.api.nvim_get_vvar('event')
+    local window_ids = event.windows
+    for stored_winid, window_context in
+      pairs(require('treesitter-context.render').get_window_contexts())
+    do
+      for _, window_id in pairs(window_ids) do
+        if stored_winid == window_id then
+          throttle(function()
```

Comment:
This doesn't work for anonymous functions since it is only ever called once.

Reply:
Do you mean throttle will collapse multiple calls into one call, if so, I should directly call the anonymous functions?

Reply:
Calling throttle like this (`throttle(...)()`) doesn't work. It needs to be.

```lua
local f = throttle(...)
f()
f() -- throttled
```

```lua
throttle(...)()
throttle(...)() -- not throttled!
```